### PR TITLE
Milestone 5: Controllers specs

### PR DESCRIPTION
**Pull request by Rachelwebdev**
### In this milestone, I:

- Added RSpec as a gem to my project.

- Created a Request spec file for all my controllers.

- Make sure that for each action I checked:

1.  If response status was correct.

2. If a correct template was rendered.

3. If the response body includes correct placeholder text.

- Checked and fixed rubocop errors

### CODE REVIEW

File: spec/requests/posts_spec.rb [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+#       expect(response).to render_template(:show)
+#     end
+
+#     it 'includes the correct text in the response body' do
+#       post = Post.create(title: 'Test Title', id: 1)
+#       get user_post_path(user_id: 1, id: post.id)
+#       expect(response.body).to include('This is a single post')
+#     end
+#   end
+# end
```
Comment (julie-ify) [inline spec/requests/posts_spec.rb]: - [ ] Kindly remove these commented code lines

File: spec/requests/posts_spec.rb [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+describe 'GET all posts for an user' do
+  let(:user) { User.create!(name: 'Test User', post_count: 0) }
+
+  it 'returns a successful response' do
+    get "/users/#{user.id}/posts"
+
+    expect(response).to be_successful
+    expect(response.body).to include('<h1>Here is a list of posts for a given user</h1>')
+    expect(response).to render_template(:index)
+  end
```
Comment (julie-ify) [inline spec/requests/posts_spec.rb:12]: - [ ] **[OPTIONAL]** Apart from writing tests that meet the expectations we also want to write tests that are descriptive. Kindly divide these tests into three 'it' blocks describing what each expectation checks for. example:

  ```
  it 'returns a successful response' do
    ...
    expect(response).to be_successful
  end
 
  it 'renders the index template' do
    ...
    expect(response).to render_template(:index)
  end
  ...
  ```

File: spec/requests/posts_spec.rb [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  let(:user) { User.create!(name: 'Test User', post_count: 0) }
+
+  it 'returns a successful response' do
+    post = Post.create!(title: 'Test Post', author_id: user.id, comments_counter: 0, likes_counter: 0)
+    get "/users/#{user.id}/posts/#{post.id}"
+
+    expect(response).to be_successful
+    expect(response.body).to include('<h1>This is a single post</h1>')
+    expect(response).to render_template(:show)
+  end
```
Comment (julie-ify) [inline spec/requests/posts_spec.rb:25]: - [ ] **[OPTIONAL]** Apart from writing tests that meet the expectations we also want to write tests that are descriptive. Kindly divide these tests into three 'it' blocks describing what each expectation checks for.

File: spec/requests/users_spec.rb [hunk 4]
```diff
@@ -0,0 +1,55 @@
+require 'rails_helper'
+
+describe 'GET /users' do
+  it 'returns a successful response' do
+    get '/users'
+
+    expect(response).to be_successful
+    expect(response.body).to include('<h1>This is a list of all Users</h1>')
+    expect(response).to render_template(:index)
+  end
```
Comment (julie-ify) [inline spec/requests/users_spec.rb:10]: - [ ] **[OPTIONAL]** Apart from writing tests that meet the expectations we also want to write tests that are descriptive. Kindly divide these tests into three 'it' blocks describing what each expectation checks for.

File: spec/requests/users_spec.rb [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+describe 'GET /users/:id' do
+  it 'returns a successful response' do
+    user = User.create!(name: 'Test User', post_count: 0)
+    get "/users/#{user.id}"
+
+    expect(response).to be_successful
+    expect(response.body).to include('<h1>This is a single user</h1>')
+    expect(response).to render_template(:show)
+  end
```
Comment (julie-ify) [inline spec/requests/users_spec.rb:21]: - [ ] **[OPTIONAL]** Apart from writing tests that meet the expectations we also want to write tests that are descriptive. Kindly divide these tests into three 'it' blocks describing what each expectation checks for.

File: spec/requests/users_spec.rb [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+#       expect(response).to render_template(:show)
+#     end
+
+#     it 'includes the correct text in the response body' do
+#       user = User.create(name: 'Chris', id: '001')
+#       get user_path(user)
+#       expect(response.body).to include('This is a single user')
+#     end
+#   end
+# end
```
Comment (julie-ify) [inline spec/requests/users_spec.rb]: - [ ] Kindly remove commented code.

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @Rachelwebdev,

Good job so far!

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by pinkmoon25 — APPROVED** [review]
Hi @Rachelwebdev , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
Linters :heavy_check_mark: 
Git flow :heavy_check_mark: 
Changes Implemented :heavy_check_mark:

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**